app/scripts/metamask-controller.js

```diff
@@ -578,6 +592,10 @@ export default class MetamaskController extends EventEmitter {
       },
       state: initState.TokensController,
     });
+    this.controllerMessenger.registerActionHandler(
+      'TokensController:getState',
+      () => this.tokensController.state,
+    );
```

Comment:
This will become redundant once `TokensController` is upgraded to extend from `BaseControllerV2`, which implicitly registers a handler for the `getState` action during the `super()` call.